### PR TITLE
Changing NuGet version to 4.0.1

### PR DESCRIPTION
**Pull request by mishra14**
Changing the NuGet version to 4.0.1-beta  on the dev branch for future. 🔥

//cc: @nkolev92 @emgarten @rohit21agrawal @jainaashish @zhili1208 @alpaix @rrelyea

### CODE REVIEW

File: test/NuGet.Clients.Tests/NuGet.VisualStudio.Implementation.Test/project.json [hunk 1]
```diff
@@ -7,12 +7,12 @@
   "frameworks": {
     "net46": {
       "frameworkAssemblies": {
-        "System.ComponentModel.DataAnnotations": "4.0.0.0"
+        "System.ComponentModel.DataAnnotations": "4.0.1.0"
```
Comment (rohit21agrawal) [inline test/NuGet.Clients.Tests/NuGet.VisualStudio.Implementation.Test/project.json]: not sure if this needs to change.

File: test/NuGet.Core.Tests/NuGet.Build.Tasks.Pack.Test/project.json [hunk 2]
```diff
@@ -25,7 +25,7 @@
       "dependencies": {
         "dotnet-test-xunit": "1.0.0-rc2-173384-19",
         "moq.netcore": "4.4.0-beta8",
-        "System.Diagnostics.TraceSource": "4.0.0",
+        "System.Diagnostics.TraceSource": "4.0.1",
```
Comment (rohit21agrawal) [inline test/NuGet.Core.Tests/NuGet.Build.Tasks.Pack.Test/project.json]: probably doesn't need to change.

File: test/NuGet.Core.Tests/NuGet.Commands.Test/project.json [hunk 3]
```diff
@@ -25,7 +25,7 @@
       "dependencies": {
         "dotnet-test-xunit": "1.0.0-rc2-173384-19",
         "moq.netcore": "4.4.0-beta8",
-        "System.Diagnostics.TraceSource": "4.0.0",
+        "System.Diagnostics.TraceSource": "4.0.1",
```
Comment (rohit21agrawal) [inline test/NuGet.Core.Tests/NuGet.Commands.Test/project.json]: same here.

File: test/NuGet.Core.Tests/NuGet.Common.Test/project.json [hunk 4]
```diff
@@ -20,8 +20,8 @@
         "moq.netcore": "4.4.0-beta8",
         "dotnet-test-xunit": "1.0.0-rc2-173384-19",
         "System.Threading.Tasks.Parallel": "4.0.1",
-        "System.Threading.Thread": "4.0.0",
-        "System.Diagnostics.TraceSource": "4.0.0",
+        "System.Threading.Thread": "4.0.1",
```
Comment (rohit21agrawal) [inline test/NuGet.Core.Tests/NuGet.Common.Test/project.json]: don't need to change.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/project.json [hunk 5]
```diff
@@ -21,7 +21,7 @@
     }
   },
   "dependencies": {
-    "System.Runtime.InteropServices.RuntimeInformation": "4.0.0",
+    "System.Runtime.InteropServices.RuntimeInformation": "4.0.1",
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/project.json]: This one stays 4.0.0 too

**Review by nkolev92 — APPROVED** [review]

**Comment by mishra14** [comment]
🔔

**Review by alpaix — APPROVED** [review]

**Review by rohit21agrawal — APPROVED** [review]

**Pull request merged**